### ISSUE TITLE
Calling snapToItem results in: undefined is not an object (evaluating 'this._venueCarousel.snapToItem')

### ISSUE THREAD

**Issue by ragunaru** [archriss/react-native-snap-carousel#301]
Hi folks,

I'm trying to snap to a new item based on a condition. 

My carousel setup:

```
      <Carousel
        ref={(venueCarousel) => { this._venueCarousel = venueCarousel; }}
        data={hits}
        renderItem={({index, item}) => (
          <MapVenueCard
             venue={item}
             index={index}
           />
        )}
        sliderWidth={width}
        itemWidth={itemWidth}
        containerCustomStyle={MapScreenStyles.VenueCardContainer}
        inactiveSlideScale={1}
        inactiveSlideOpacity={1}
        onSnapToItem={this.onSnap}
      />
```

I have this function that uses the snapToItem callback.

```
  resetCards(activeVenueCard) {
    this._venueCarousel.snapToItem(activeVenueCard, true);
  }
```

And then I call that function in an if statement:

```
    if (activeVenueCard == 0) {
      this.resetCards(activeVenueCard);
    } else {
      console.log('============ Tings are not 0');
    }
```

But I keep getting this error:

![image](https://user-images.githubusercontent.com/795389/38458715-5458455c-3a6f-11e8-9169-eebf00a26774.png)

The strange thing is if I wrap the callback in a setTimeout, it works just fine!

```
  resetCards(activeVenueCard) {
    setTimeout(() => {
      this._venueCarousel.snapToItem(activeVenueCard, true);
    }, 500);
  }
```

Why could this be? Any help appreciated.

**Comment by bd-arc** [comment]
@ragunaru My guess is that `onSnapToItem` is called during init before the reference has been created.

Try the following:
```
resetCards(activeVenueCard) {
  this._venueCarousel && this._venueCarousel.snapToItem(activeVenueCard, true);
}
```

**Comment by bd-arc** [comment]
@ragunaru This should be fixed in version `3.7.0`.

**Comment by jhoanborges** [comment]
Bin the render function in the constructor 
`  constructor(props) {

    super(props);
    this._renderItem = this._renderItem.bind(this);`
